src/synology_dsm/api/core/external_usb.py

```diff
@@ -0,0 +1,245 @@
+                self._data[device["dev_id"]] = SynoCoreExternalUSBDevice(device)
+
+    # Root
+    @property
+    def get_devices(self):
+        """Gets all external USB storage devices."""
+        return self._data
+
+    # Device
+    @property
+    def device_ids(self):
+        """Returns external USB storage device ids."""
+        devices = []
+        for device in self.get_devices:
+            devices.append(device)
+        return devices
+
+    def get_device(self, device_id) -> SynoCoreExternalUSBDevice:
+        """Returns a specific external USB storage device."""
+        return self._data[device_id]
+
+
+class SynoCoreExternalUSBDevice:
+    """A representation of an external USB device."""
+
+    def __init__(self, data):
+        """Initialize a external USB device."""
+        self._data = data
+        for partition_data in data["partitions"]:
+            self._data[partition_data["name_id"]] = SynoUSBStoragePartition(
+                partition_data
+            )
+
+    @property
+    def device_id(self):
+        """Return id of the device."""
+        return self._data["dev_id"]
+
+    @property
+    def device_name(self):
+        """The title of the external USB storage device."""
+        return self._data["dev_title"]
+
+    @property
+    def device_type(self):
+        """The type of the external USB storage device."""
+        return self._data["dev_type"]
+
+    def device_size_total(self, human_readable=False):
+        """Total size of the external USB storage device."""
+        return_data = int(self._data["total_size_mb"])
+        return_data = SynoFormatHelper.megabytes_to_bytes(return_data)
+        if human_readable:
+            return SynoFormatHelper.bytes_to_readable(return_data)
+        return return_data
+
+    @property
+    def device_status(self):
+        """The status of the external USB storage device."""
+        return self._data["status"]
+
+    @property
+    def device_formatable(self):
+        """Whether the external USB storage device can be formatted."""
+        return self._data["formatable"]
+
+    @property
+    def device_progress(self):
+        """The progress the external USB storage device."""
+        return self._data["progress"]
+
+    @property
+    def device_product_name(self):
+        """The product name of the external USB storage device."""
+        return self._data["product"]
+
+    @property
+    def device_manufacturer(self):
+        """The producer name of the external USB storage device."""
+        return self._data["producer"]
+
+    # Partition
+    @property
+    def device_partitions(self) -> list[SynoUSBStoragePartition]:
+        """Returns all partitions of the external USB storage device."""
+        partitions = self._data["partitions"]
+        device_partitions: list[SynoUSBStoragePartition] = []
+        for partition in partitions:
+            device_partitions.append(SynoUSBStoragePartition(partition))
+        return device_partitions
```

Comment:
do just `return self._data["partitions"]`, because it is already parsed and casted during `__init__`
return type will be `dict[str, SynoUSBStoragePartition]`